namespaces/mson-namespace.md

```diff
@@ -0,0 +1,497 @@
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "member",
+            "content": {
+                "key": {
+                    "element": "string",
+                    "content": "id"
+                },
+                "value": {
+                    "element": "number",
+                    "attributes": {
+                        "default": 0
+                    },
+                    "content": null
+                }
+            }
+        }
+    ]
+}
+```
+
+### One Of
+
+#### MSON
+
+```
+- city
+- One Of
+    - state
+    - province
+```
+
+#### MSON DOM
+
+TODO:
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "member",
+            "content": {
+                "key": {
+                    "element": "string",
+                    "content": "city"
+                }
+            }
+        },
+        {
+            "element": "select",
+            "content": [
+                {
+                    "element": "option",
+                    "content": [
+                        {
+                            "element": "member",
+                            "content": {
+                                "key": {
+                                    "element": "string",
+                                    "content": "state"
+                                }
+                            }
+                        }
+                    ]
+                },
+                {
+                    "element": "option",
+                    "content": [
+                        {
+                            "element": "member",
+                            "content": {
+                                "key": {
+                                    "element": "string",
+                                    "content": "province"
+                                }
+                            }
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+}
+```
+
+### Mixin
+
+#### MSON
+
+```
+- id
+- Include User
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "extend",
+    "content": [
+        {
+            "element": "object",
+            "content": [
+                {
+                    "element": "member",
+                    "content": {
+                        "key": {
+                            "element": "string",
+                            "content": "id"
+                        }
+                    }
+                }
+            ]
+        }
+        {
+            "element": "ref",
+            "content": "User"
+        }
+    ]
+}
+```
+
+### Named Type
+
+#### MSON
+
+```
+# Address (object)
+
+Description is here! Properties to follow.
+
+## Properties
+
+- street
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "meta": {
+        "id": "Address",
+        "title": "Address",
+        "description": "Description is here! Properties to follow."
```

Comment:
So, in MSON there is a block description. If you put description here, then there is no way to differentiate an inline description and round trip back to mson. Also, I think we can define block descriptions and groups as a member since and MSON element is bascially an object. The thing is that if we put things in attributes, we lose track of the structure of the associated MSON document. I don't think this is good. Am I making sense @zdne?